neard/src/runtime.rs

```diff
@@ -892,6 +892,39 @@ impl RuntimeAdapter for NightshadeRuntime {
         Ok(epoch_start_height)
     }
 
+    fn check_sync_hash_on_epoch_boundary(&self, sync_hash: &CryptoHash) -> bool {
```

Comment:
you can pass `last_block_hash` here to check the recency of `sync_hash`.

Reply:
++